### PR TITLE
Relocate GQL Data Loaders

### PR DESCRIPTION
**Pull request by tn3rb**
see #2708

- adds DataLoaderManager and supporting classes for registering data loaders for the event editor domain

- this can be iterated upon in the future to accommodate additional domains and load dependencies dynamically based on current route

### CODE REVIEW

File: core/domain/services/graphql/data/domains/EspressoEditor.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public function registerLoaders($loaders, $context)
+    {
+        $newLoaders = [
+            'espresso_attendee'  => new Loaders\AttendeeLoader($context),
+            'espresso_datetime'  => new Loaders\DatetimeLoader($context),
+            'espresso_price'     => new Loaders\PriceLoader($context),
+            'espresso_priceType' => new Loaders\PriceTypeLoader($context),
+            'espresso_ticket'    => new Loaders\TicketLoader($context),
+            'espresso_venue'     => new Loaders\VenueLoader($context),
+        ];
```
Comment (manzoorwanijk) [inline core/domain/services/graphql/data/domains/EspressoEditor.php:36]: Cool 👍

File: core/EE_Dependency_Map.core.php [hunk 2]
```diff
@@ -911,10 +911,11 @@ protected function _register_core_dependencies()
...
             'EventEspresso\core\services\graphql\GraphQLManager' => [
-                'EventEspresso\core\services\graphql\TypesManager'  => EE_Dependency_Map::load_from_cache,
-                'EventEspresso\core\services\graphql\InputsManager'  => EE_Dependency_Map::load_from_cache,
-                'EventEspresso\core\services\graphql\EnumsManager'  => EE_Dependency_Map::load_from_cache,
-                'EventEspresso\core\services\graphql\ConnectionsManager'  => EE_Dependency_Map::load_from_cache,
+                'EventEspresso\core\services\graphql\ConnectionsManager' => EE_Dependency_Map::load_from_cache,
+                'EventEspresso\core\services\graphql\DataLoaderManager'  => EE_Dependency_Map::load_from_cache,
+                'EventEspresso\core\services\graphql\EnumsManager'       => EE_Dependency_Map::load_from_cache,
+                'EventEspresso\core\services\graphql\InputsManager'      => EE_Dependency_Map::load_from_cache,
+                'EventEspresso\core\services\graphql\TypesManager'       => EE_Dependency_Map::load_from_cache,
```
Comment (nicolad) [inline core/EE_Dependency_Map.core.php:918]: ABCDEF ... OMG :+1:

File: core/domain/services/graphql/data/domains/EspressoEditor.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    public function registerLoaders($loaders, $context)
+    {
+        $newLoaders = [
+            'espresso_attendee'  => new Loaders\AttendeeLoader($context),
+            'espresso_datetime'  => new Loaders\DatetimeLoader($context),
+            'espresso_price'     => new Loaders\PriceLoader($context),
+            'espresso_priceType' => new Loaders\PriceTypeLoader($context),
+            'espresso_ticket'    => new Loaders\TicketLoader($context),
+            'espresso_venue'     => new Loaders\VenueLoader($context),
```
Comment (nicolad) [inline core/domain/services/graphql/data/domains/EspressoEditor.php:35]: feast for my :eyes:

File: core/services/graphql/loaders/DataLoaderCollection.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function __construct()
+    {
+        parent::__construct(
+            DataLoaderCollection::INTERFACE,
+            DataLoaderCollection::NAME
+        );
+    }
+
+
```
Comment (nicolad) [inline core/services/graphql/loaders/DataLoaderCollection.php:47]: so much space here, need to tweak prettier to allow also 2 lines of empty spaces in JS :)

File: core/services/graphql/loaders/DataLoaderCollection.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        if (! $this->loader instanceof CollectionLoader) {
+            $this->loader = new CollectionLoader(
+                new CollectionDetails(
+                    // collection name
+                    DataLoaderCollection::NAME,
+                    // collection interface
+                    DataLoaderCollection::INTERFACE,
+                    // FQCNs for classes to add (all classes within each namespace will be loaded)
+                    apply_filters(
+                        'FHEE__EventEspresso_core_services_graphql_DataLoaderCollection__loadCollection__collection_FQCNs',
```
Comment (nicolad) [inline core/services/graphql/loaders/DataLoaderCollection.php:64]: this is one word?
I got lost somewhere in the middle

File: core/services/graphql/loaders/DataLoaderCollection.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                        ['EventEspresso\core\domain\services\graphql\data\domains']
+                    ),
+                    // filepaths to classes to add
+                    array(),
+                    // file mask to use if parsing folder for files to add
+                    '',
+                    // what to use as identifier for collection entities
+                    // using CLASS NAME prevents duplicates (works like a singleton)
+                    CollectionDetails::ID_CLASS_NAME
```
Comment (nicolad) [inline core/services/graphql/loaders/DataLoaderCollection.php:73]: nice comments :+1:

File: core/services/graphql/loaders/DataLoaderCollection.php (hunk 5 above)
Comment (tn3rb) [reply core/services/graphql/loaders/DataLoaderCollection.php:64]: yup:

 - `FHEE`: **F**ilter **H**ook **E**vent **E**sspresso

 - `EventEspresso_core_services_graphql_DataLoaderCollection`: fully qualified classname where hook is located

 - `loadCollection`: method where hook is located

 - `collection_FQCNs`: the thing being filtered

Comment (nicolad) [reply core/services/graphql/loaders/DataLoaderCollection.php:64]: thanks for explanation :bowing_man:

**Pull request merged**